### ISSUE TITLE
`#[deny(unreachable_patterns)]` does not yield an error with duplicated endpoints

### ISSUE THREAD

**Issue by glaeqen** [jamesmunns/postcard-rpc#34]
# To reproduce:

1. Apply patch
```diff
diff --git a/example/firmware/src/bin/comms-02.rs b/example/firmware/src/bin/comms-02.rs
index ad1efff..b7b4106 100644
--- a/example/firmware/src/bin/comms-02.rs
+++ b/example/firmware/src/bin/comms-02.rs
@@ -68,6 +68,7 @@ define_dispatch! {
         Context = Context,
     >;
     PingEndpoint => blocking ping_handler,
+    PingEndpoint => blocking ping_handler,
     GetUniqueIdEndpoint => blocking unique_id_handler,
     SetSingleLedEndpoint => async set_led_handler,
     SetAllLedEndpoint => async set_all_led_handler,
```
2. Try to build `firmware` crate
3. Observe success

# Expected behaviour
Compilation failure

**Comment by jamesmunns** [comment]
This seems to be down to the fact that lints in expanded code doesn't fire.

If I make one crate that has this:

```rust
pub trait Val {
    const VAL: [u8; 8];
}

impl Val for u8 {
    const VAL: [u8; 8] = [0u8; 8];
}

impl Val for u16 {
    const VAL: [u8; 8] = [1u8; 8];
}


#[macro_export]
macro_rules! example {
    ($val: ident, $($name:ty),*) => {
        // We want this macro to fire if there are dupes
        #[deny(unreachable_patterns)]
        match $val {
            $(
                <$name as Val>::VAL => println!("$name"),
            )*
            _ => println!("Other"),
        }
    };
}
```

And another crate that does this:

```rust
#![deny(unreachable_patterns)]

use macro_source::Val;

fn main() {
    let val = [2u8; 8];
    macro_source::example!(val, u8, u8, u16);
}
```

I don't get the lint to fire at all

**Comment by jamesmunns** [comment]
Asked on T-Lang: https://rust-lang.zulipchat.com/#narrow/stream/213817-t-lang/topic/deny.20lint.20behavior.20in.20macro.20expansion/near/444722399

**Comment by jamesmunns** [comment]
Addressed by #35, released in v0.5.1
